src/SFA.DAS.AssessorService.Application.Api.External/Infrastructure/ApiClient.cs

```diff
@@ -60,19 +110,69 @@ protected async Task<T> Get<T>(string uri)
             var serializeObject = JsonConvert.SerializeObject(model);
 
-            using (var response = await _client.PutAsync(new Uri(uri, UriKind.Relative), new StringContent(serializeObject, System.Text.Encoding.UTF8, "application/json")))
+            try
             {
-                return await response.Content.ReadAsAsync<U>();
+                using (var response = await _client.PutAsync(new Uri(uri, UriKind.Relative), new StringContent(serializeObject, System.Text.Encoding.UTF8, "application/json")))
+                {
+                    try
+                    {
+                        return await response.Content.ReadAsAsync<U>();
+                    }
+                    catch (Exception ex)
+                    {
+                        var actualResponse = string.Empty;
+                        try
+                        {
+                            actualResponse = await response.Content.ReadAsStringAsync();
+                        }
+                        catch
+                        {
+                            // safe to ignore any errors
+                        }
+                        _logger.LogError(ex, $"PUT: HTTP {(int)response.StatusCode} Error getting response from: {uri} - ActualResponse: {actualResponse}");
+                        throw;
+                    }
+                }
+            }
+            catch (HttpRequestException ex)
+            {
+                _logger.LogError(ex, $"PUT: HTTP Error when processing request to: {uri}");
+                throw;
             }
         }
 
         protected async Task<T> Delete<T>(string uri)
         {
             _client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", _tokenService.GetToken());
 
-            using (var response = await _client.DeleteAsync(new Uri(uri, UriKind.Relative)))
+            try
+            {
+                using (var response = await _client.DeleteAsync(new Uri(uri, UriKind.Relative)))
+                {
+                    try
+                    {
+                        return await response.Content.ReadAsAsync<T>();
+                    }
+                    catch (Exception ex)
+                    {
+                        var actualResponse = string.Empty;
+                        try
+                        {
+                            actualResponse = await response.Content.ReadAsStringAsync();
+                        }
+                        catch
+                        {
+                            // safe to ignore any errors
+                        }
+                        _logger.LogError(ex, $"DELETE: HTTP {(int)response.StatusCode} Error getting response from: {uri} - ActualResponse: {actualResponse}");
+                        throw;
+                    }
+                }
+            }
+            catch (HttpRequestException ex)
```

Comment:
Can this error handling code (that looks the same in each instance) be refactored out?

Reply:
They are HTTP method specific so I think it'll leave a messy refactored method